src/secondary.rs

```diff
@@ -135,6 +135,20 @@ where
         self.data.capacity()
     }
 
+    /// Remove key `old` and optionally move to key `new`.
+    ///
+    /// This method is useful for rekey callbacks such as in
+    /// [`PortGraph::set_num_ports`] and [`PortGraph::compact_nodes`].
+    pub fn rekey(&mut self, old: K, new: Option<K>)
+    where
+        V: Default,
+    {
+        let val = mem::take(self.get_mut(old));
+        if let Some(new) = new {
+            self[new] = val;
```

Comment:
This needs to call `resize_for_get_mut` first, since `new` may not have been allocated.
Check the `swap` function, which does something similar.

Reply:
(Maybe move put this and `swap` together, for readability)